service/history/transferQueueProcessorBase.go

```diff
@@ -163,9 +165,26 @@ func (t *transferQueueProcessorBase) recordWorkflowStarted(
 		return nil
 	}
 
+	var memo []byte
+	encoding := t.shard.GetEncoding(domainEntry)
+	memoBlob, err := t.serializer.SerializeVisibilityMemo(visibilityMemo, encoding)
```

Comment:
This is incorrect layer for serialization.

Reply:
I agree, this also requires bigger refactor of es visibility manager. Would you mind me do it in another PR?